modules/docs/mdx/9.0-UPGRADE-GUIDE.mdx

```diff
@@ -52,133 +61,164 @@ handled by the codemod are marked with 🤖 in the Upgrade Guide.**
 
-We found an issue where the default action for `form` controls like `input` and `textarea` is to
-click the first `submit` `button` which in some cases is not a `submit` button.
+Any buttons which extend any of the above are affected as well.
 
-An example of this causing issues was our `Modal.CloseIcon` and `Modal.CloseButton`. When the
-`Modal` is wrapped in a `form` element, the close icon would submit the `form`.
+This resolves an [issue](https://github.com/Workday/canvas-kit/issues/1938) where clicking certain
+buttons within a `form` element would unexpectedly submit the form. Additionally, the default action
+of many form controls such as `input` and `textarea` is to click the first submit button; this led
+to issues if the first submit button in the form was not intended to be a submit button (an easy
+mistake to make when buttons default to `type="submit"`).
 
-This change affects all of our `buttons` including:
-`PrimaryButton, SecondaryButton, TertiaryButton, ToolbarButton`.
+This is a breaking change if you expected a button to submit by default, though we do not anticipate
+this to be the case for most developers. If you _do_ intend for a button to act as a submit button
+within a form, add `type="submit"` to the button.
 
-If you still wish for a specific `button` within a `form` to submit, just add the attribute
-`type="submit"`.
+---
 
 ### Toast
 
-We've promoted `Toast` from Labs (`@workday/canvas-kit-labs-react`) to the Main
-(`@workday/canvas-kit-react`) package.
+We've promoted `Toast` from Labs to Main. It replaces the `Toast` that was previously in Main (for
+reference, see
+[`Toast`](https://d2krrudi3mmzzw.cloudfront.net/v8/?path=/docs/components-popups-toast--error) in
+Main from v8).
 
-- This component is a
-  [compound component](/getting-started/for-developers/resources/compound-components/).
-- It has a `mode` prop that allows for proper accessibility attributes to be defined.
-- Its sub components allow for more flexibility and access to lower level elements.
+`Toast` is a [compound component](/getting-started/for-developers/resources/compound-components/)
+which provides a flexible API and access to its internals via its subcomponents. It supports a new
+`mode` prop which applies the proper accessibility features to the component based on the desired
+mode: `status`, `alert`, or `dialog`.
 
 ```tsx
 // v8
-<Toast actionText="View more details" onClose={handleClose}>Your workbook was successfully processed.</Toast>
+<Toast actionText="View more details" onActionClick={handleActionClick} onClose={handleClose}>
+  Your workbook was successfully processed.
+</Toast>
 
 // v9
 <Toast mode="dialog">
   <Toast.Icon icon={checkIcon} color={colors.greenApple400} />
   <Toast.Body>
     <Toast.Message>Your workbook was successfully processed.</Toast.Message>
-    <Toast.Link href="#hreflink">Custom Link</Toast.Link>
+    <Toast.Link onActionClick={handleActionClick}>View more details</Toast.Link>
```

Comment:
`onActionClick` use to be an prop, we've removed this, maybe this would be `onClick`

Reply:
Good catch. I modeled this snippet off your compoundToast codemod test [here](https://github.com/Workday/canvas-kit/blob/prerelease/major/modules/codemod/lib/v9/spec/compoundToast.spec.ts#L83), but I got this detail wrong.

On this topic, your original Toast snippet here assigns `aria-label="Close"` to `Toast.CloseIcon` but the expected output in the test I linked above doesn't have `aria-label`. Should I keep the `aria-label` here?

Reply:
Hmm ok, I did some investigating. All of our examples have an `aria-label` but `aria-label` is not added by the component itself. I'm how to handle this in the codemod. I don't want to overwrite something that might have been translated. Let me think about this one

Reply:
We decided _not_ to update the codemod to handle `aria-label` (to avoid potential translation issues) but to leave the `aria-label` in the code snippet to encourage proper a11y. We'll mention adding `aria-label` to `Toast.CloseIcon` in the Toast component documentation.